packages/vite/src/node/plugins/worker.ts

```diff
@@ -75,13 +75,14 @@ async function bundleWorkerEntry(
   // bundle the file as entry to support imports
   const { rollup } = await import('rollup')
   const { plugins, rollupOptions, format } = config.worker
-  const { plugins: resolvedPlugins, config: workerConfig } =
-    await plugins(newBundleChain)
+  const workerConfig = await plugins(newBundleChain)
   const workerEnvironment = new BuildEnvironment('client', workerConfig) // TODO: should this be 'worker'?
+  await workerEnvironment.init()
```

Comment:
IIUC this line is the main fix, and the rest are refactors?

Reply:
Also updating config with new plugins like `{ ...config, plugins: resolvedPlugins }` is needed before init.